daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/PhaseOne.scala

```diff
@@ -274,7 +274,7 @@ private[lf] final class PhaseOne(
       case EThrow(_, ty, e) =>
         SBThrow(SBToAny(ty)(compile(env, e)))
       case EToInterface(iface @ _, tpl @ _, e) =>
-        compile(env, e) // interfaces have the same representation as underlying template
+        SBToInterface(tpl)(compile(env, e))
       case EFromInterface(iface @ _, tpl, e) =>
         SBFromInterface(tpl)(compile(env, e))
```

Comment:
I think this as nothing to do with deriving `Eq` intance ?

Reply:
yes sorry that's on me since I've also rebased the other branch. I'll set this PR back to draft, sorry for the confusion

Reply:
Nothing to be sorry about. :smiley:

Reply:
yes will do, I was planning for that to be done first thing today but CI happened :(